go/libraries/doltcore/diff/table_deltas.go

```diff
@@ -270,12 +270,10 @@ func matchTableDeltas(fromDeltas, toDeltas []TableDelta) (deltas []TableDelta) {
 	for _, name := range matchedNames {
```

Comment:
(minor) Might be worth a comment for this block and possibly the similar block starting on line 281 to help document the intent of this code. We were using `schemasOverlap` in both blocks, but we're taking it out of just this one. It's probably a little hard to reverse engineer the meaning behind that, so adding a little context could be helpful in the future the next time someone goes into this code. It seems like when we match a table by name, we want to skip the column tag match, but when we don't match by name, we want to try and be clever and see if they are the same tags (so we can try to catch a table rename presumably). Would be nice to leave some of that context about why the code works this way to help document the expected behavior.